source/command-monitoring/tests/unified/redacted-commands.yml

```diff
@@ -0,0 +1,166 @@
+description: "redacted-commands"
+
+schemaVersion: "1.0"
+
+runOnRequirements:
+  - minServerVersion: "5.0"
+
+createEntities:
+  - client:
+      id: &client client
+      observeEvents:
+        - commandStartedEvent
+  - database:
+      id: &database database
+      client: *client
+      databaseName: &databaseName command-monitoring-tests
+
+tests:
+  - description: "authenticate"
+    operations:
+      - name: runCommand
+        object: *database
+        arguments:
+          commandName: authenticate
+          command:
+            authenticate: "private"
+        # Malformed authentication commands will fail against the server, but we
+        # just want to assert that security-related commands are redacted
+        # in command monitoring.
+        expectError:
+          isError: true
+    expectEvents: &redacted-event
+      - client: *client
+        events:
+          # Expect empty (redacted) event.
+          - commandStartedEvent: {}
```

Comment:
[Security](https://github.com/mongodb/specifications/blob/master/source/command-monitoring/command-monitoring.rst#security) in the spec states that the command and/or reply documents are supposed to be redacted. In this case, I think it makes sense to have `commandStartedEvent` use `commandName` to assert a particular command was observed and _then_ use `$$exists: false` on a field within the command/reply body that would otherwise be expected. For instance:

```
- commandStartedEvent:
    commandName: redactedCommandName
    command: { redactedCommandName: { $$exists: false } }
```

Reply:
That makes sense, thanks for the explanation here and offline! Changed the expected event to be
```
- commandStartedEvent:
    commandName: redactedCommandName
    command: { redactedCommandName: { $$exists: false } }
```
And left a comment similar to the one above. I agree that it's unlikely a driver would redact only the command name field from the published `command`. I _could_ add further `{ $$exists: false }` assertions for other fields like `mechanism` and `payload` if that level of detail seems important to others.

Reply:
for the `command` assertion, I assume we need to do it that way (with `$$exists`) because commands are considered [root documents](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#allowing-extra-fields-in-root-level-documents) and allowed to have extra fields?

this seems fine to me, as I think adding to the format just for this to support e.g. an empty doc check would probably be overkill. but I don't think it's necessarily obvious that the test is only written in this less optimal way because of that restriction. perhaps worth mentioning in the comment as well?

> I _could_ add further `{ $$exists: false }` assertions for other fields like `mechanism` and `payload` if that level of detail seems important to others

I agree with @jmikola that it's unlikely someone would remove the command name but omit the other fields, so IMO this isn't needed

Reply:
Yep, I think that section about root documents is why we can't just say `command: {}`. I expanded on the comment to explain that restriction. I think @jyemin agreed offline that adding to the format for this situation would not have much benefit for us in the future.